docs/app/documentation/component-docs/date-picker/examples/date-picker-format-example.component.ts

```diff
@@ -0,0 +1,60 @@
+import { Component, Injectable } from '@angular/core';
+import { DateFormatParser } from '../../../../../../library/src/lib/calendar/format/date-parser';
+
+@Injectable()
+export class DateFormatDashes extends DateFormatParser {
+
+    rangeDelimiter: string = ' to ';
+
+    public parse(value: string): Date {
+        const values: number[] = value.split('-').map(Number);
+
+        // If date is 0, set the date to invalid by setting month to 14
+        if (values[2] === 0) {
+            values[1] = 14;
+        }
+        return new Date(Number(values[0]), values[1] - 1, values[2]);
+    }
+
+    public format(date: Date): string {
+        let monthStr = (date.getMonth() + 1).toLocaleString();
+        if (monthStr.length === 1) {
+            monthStr = '0' + monthStr;
+        }
+
+        return date.getFullYear() + '-' + monthStr + '-' + date.getDate();
```

Comment:
I think it's a good idea to have more details/comments about this line of code and the whole `format` function in order to avoid any confusion for the user.